addons/payment_razorpay/models/payment_provider.py

```diff
@@ -214,20 +214,28 @@ def _razorpay_make_request(self, endpoint, payload=None, method='POST', api_vers
             )
         return response.json()
 
-    def _razorpay_calculate_signature(self, data):
+    def _razorpay_calculate_signature(self, data, is_redirect=True):
         """ Compute the signature for the request's data according to the Razorpay documentation.
 
         See https://razorpay.com/docs/webhooks/validate-test#validate-webhooks.
 
         :param bytes data: The data to sign.
+        :param bool is_redirect: Whether the data should be treated as redirect data or as coming
+                                 from a webhook notification.
         :return: The calculated signature.
         :rtype: str
         """
-        secret = self.razorpay_webhook_secret
+
+        secret = self.razorpay_key_secret if is_redirect else self.razorpay_webhook_secret
```

Comment:
razorpay_key_secret will always be false for Oauth process so it'll always skip this check :thinking: